### PR TITLE
MOVE-187: Added tests.

### PR DESCRIPTION
**Pull request by martinwam**

### CODE REVIEW

**Comment by nuth** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR no.difi.meldingsutveksling:serviceregistry-parent: Filename: spring-cloud-starter-ribbon-1.1.0.RELEASE.jar | Reference: CVE-2016-9878 | CVSS Score: 5.0 | Category: CWE-22 Improper Limitation of a Pathname to a Restricted Directory ('Path Traversal') | An issue was discovered in Pivotal Spring Framework before 3.2.18, 4.2.x before 4.2.9, and 4.3.x before 4.3.5. Paths provided to the ResourceServlet were not properly sanitized and as a result exposed to directory traversal attacks. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. MAJOR no.difi.meldingsutveksling:serviceregistry-parent: Filename: spring-cloud-starter-1.1.0.RELEASE.jar | Reference: CVE-2016-9878 | CVSS Score: 5.0 | Category: CWE-22 Improper Limitation of a Pathname to a Restricted Directory ('Path Traversal') | An issue was discovered in Pivotal Spring Framework before 3.2.18, 4.2.x before 4.2.9, and 4.3.x before 4.3.5. Paths provided to the ResourceServlet were not properly sanitized and as a result exposed to directory traversal attacks. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)

File: serviceregistry-server/pom.xml [hunk 1]
```diff
@@ -205,6 +205,11 @@
         </dependency>
 
         <dependency>
+            <groupId>org.springframework.security</groupId>
+            <artifactId>spring-security-test</artifactId>
```
Comment (mhoel) [inline serviceregistry-server/pom.xml]: Kunne lagt på en <scope>test her

File: serviceregistry-server/src/test/java/no/difi/meldingsutveksling/serviceregistry/controller/ServiceRecordControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        when(kontaktInfoMock.isReservert()).thenReturn(false);
+        PostAddress postAddress = mock(PostAddress.class);
+        PostAddress returnAddress = mock(PostAddress.class);
+        SikkerDigitalPostServiceRecord dpiServiceRecord = new SikkerDigitalPostServiceRecord(null, kontaktInfoMock, ServiceIdentifier.DPI, "12345678901", postAddress, returnAddress);
+        when(serviceRecordFactory.createServiceRecordForCititzen(eq("12345678901"), any(), any())).thenReturn
+                (dpiServiceRecord);
+    }
+
+    @Test
+    public void entityShouldReturnOK() throws Exception {
```
Comment (mhoel) [inline serviceregistry-server/src/test/java/no/difi/meldingsutveksling/serviceregistry/controller/ServiceRecordControllerTest.java]: Kanskje rename testen sånn at den har samme navngiving som testen under: eduEntityShouldReturnOk elns.

File: serviceregistry-server/src/test/java/no/difi/meldingsutveksling/serviceregistry/controller/ServiceRecordControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.serviceRecord.organisationNumber", is("42")))
+                .andExpect(jsonPath("$.serviceRecord.serviceIdentifier", is("EDU")))
+                .andExpect(jsonPath("$.serviceRecord.pemCertificate", is("pem123")))
+                .andExpect(jsonPath("$.serviceRecord.serviceCode", is("123")))
+                .andExpect(jsonPath("$.serviceRecord.serviceEditionCode", is("321")))
+                .andExpect(jsonPath("$.serviceRecord.endPointURL", is("http://foo")))
+                .andExpect(jsonPath("$.infoRecord.identifier", is("42")))
+                .andExpect(jsonPath("$.infoRecord.entityType.name", is("Organisasjonsledd")))
+                .andExpect(jsonPath("$.infoRecord.entityType.acronym", is("ORGL")));
```
Comment (mhoel) [inline serviceregistry-server/src/test/java/no/difi/meldingsutveksling/serviceregistry/controller/ServiceRecordControllerTest.java]: Kanskje ta bort noen av expect-ene og kanskje generalisert noen av jsonPath-ene så det blir mindre sjanse for å bryte testen bare man gjør en mindre refaktorering? I hvertfall kanskje splitta expect status ok og EDU i en separat test.

**Pull request closed without merging**